### PR TITLE
Set engine env from common config

### PR DESCRIPTION
**Pull request by QiWang19**
vendor commond PR https://github.com/containers/common/pull/198 to set container_engine_env

related issue  https://github.com/containers/common/issues/31

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by QiWang19** [comment]
@rhatdan @mheon  PTAL

File: cmd/podman/root.go [hunk 1]
```diff
@@ -128,6 +128,18 @@ func persistentPreRunE(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	for _, env := range cfg.Engine.Env {
+		splitEnv := strings.SplitN(env, "=", 2)
+		key := splitEnv[0]
+		value := ""
+		if len(splitEnv) == 2 {
+			value = splitEnv[1]
+		}
```
Comment (rhatdan) [inline cmd/podman/root.go:137]: If it is not 2, this should also be an error.

Comment (rhatdan) [reply cmd/podman/root.go:137]: I see what you did, but users might accidentally override an existing environement variable with nothing.
If they want nothing then they should set it 
foo=""

**Comment by rhatdan** [comment]
LGTM
@mheon @baude @giuseppe @vrothberg @TomSweeneyRedHat @ashley-cui @jwhonce PTAL

File: cmd/podman/root.go [hunk 2]
```diff
@@ -128,6 +128,16 @@ func persistentPreRunE(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	for _, env := range cfg.Engine.Env {
+		splitEnv := strings.SplitN(env, "=", 2)
```
Comment (mheon) [inline cmd/podman/root.go:132]: Do we want this to take precedence over actual OS environment? I imagine that if I have something set in containers.conf but later do a `ENVVAR=value podman run ...` I want the version passed there to take precedence.

If so, we should only define environment variables that are not already defined.

File: cmd/podman/root.go [hunk 3]
```diff
@@ -128,6 +128,16 @@ func persistentPreRunE(cmd *cobra.Command, args []string) error {
...
+		if len(splitEnv) != 2 {
+			return fmt.Errorf("invalid environment variable for engine %s, valid configuration is KEY=value pair", env)
+		}
+		if err := os.Setenv(splitEnv[0], splitEnv[0]); err != nil {
```
Comment (giuseppe) [inline cmd/podman/root.go:136]: should this be ` os.Setenv(splitEnv[0], splitEnv[1])` ?

Comment (QiWang19) [reply cmd/podman/root.go]: yes. fixed....

File: cmd/podman/root.go (hunk 2 above)
Comment (rhatdan) [reply cmd/podman/root.go:132]: Yes I guess that makes sense.  @QiWang19 Please add a note to the man pages on podman (buildah) about this, also, just do a check to see if environment is set, it set write a debug message, if different, and don't override,

**Comment by QiWang19** [comment]
@rhatdan @mheon PTAL

**Pull request merged**